1.regression_modelling/scripts/2.r2_figure.py

```diff
@@ -0,0 +1,657 @@
+sns.stripplot(
+    data=holdout_df,
+    x='R2 score',
+    y='Stain',  # Same axis for horizontal separation
+    hue='Measurement',  # Color by 'Measurement'
+    palette=palette_dict,  # Customize colors with your palette
+    jitter=True,  # Jitter the points slightly for better visibility
+    dodge=True,
+    size=8,  # Size of the scatter points
+    edgecolor='black',
+    linewidth=1,
+    alpha=0.8  # Transparency of the points
+)
+
+# Add a vertical dashed line at R2 score = 0.5
+plt.axvline(x=0.5, color='red', linestyle='--', linewidth=1)
+
+# Customize the plot
+plt.xlabel('R2 Score')
+plt.ylabel('Model Stain')
+
+# Move legend outside of the plot
+plt.legend(title='Measurement', bbox_to_anchor=(1.05, 1), loc='upper left')
+
+# Improve layout
+plt.tight_layout()
+
+# Save
+plt.savefig(f"./{figures_dir}/holdout_a647_versus_gold_r2.png", dpi=600)
+
+# Show plot
+plt.show()
+
+
+# ## Generate boxplot figure comparing the top 2 model feature distributions across stains using the training data
+# 
+# Note: We do not save these figures as the results from them do not tell a clear story to use elsewhere.
+
+# ### Identify top two features per stain
+# 
+# These will be the same two RadialDistribution features (FracatD and Zernike 2,2).
+
+# In[18]:
+
+
+# Create a list to store top features
+top_features_list = []
+
+# Iterate over each stain
+for stain in ['A647', 'GOLD']:
+    # Filter features for the current stain and sort by R2 score
+    stain_features_df = final_results_df[
+        ~final_results_df[f'{stain} R2 score'].isna()
+    ].sort_values(by=f'{stain} R2 score', ascending=False)
+    
+    # Get the top 2 features for the each stain
+    top_stain_features = stain_features_df.head(4)[f'Feature_{stain}'].tolist()
+    
+    # Add to the top features list
+    top_features_list.extend(top_stain_features)
+
+# Remove duplicates if any
+top_features_list = list(set(top_features_list))
+
+# Print the top features list
+print(len(top_features_list))
+print(top_features_list)
+
+
+# ### Filter training data to only include metadata and top feature columns
+
+# In[19]:
+
+
+# Get all columns with the 'Metadata' prefix
+metadata_columns = [col for col in training_df.columns if col.startswith('Metadata')]
+
+# Combine the lists of top features and metadata columns
+selected_columns = metadata_columns + top_features_list
+
+# Filter the training_df dataframe to include only the selected columns
+filtered_training_df = training_df[selected_columns]
+
+print(filtered_training_df.shape)
+filtered_training_df.head()
+
+
+# ### Plot distributions across features per feature
+
+# In[20]:
+
+
+# Set the figure size
+plt.figure(figsize=(12, 8))
+
+# Create the vertical box plot
+sns.boxplot(
+    data=filtered_training_df,
+    x='Metadata_Condition',  # Metadata_condition on x-axis
+    y='Nuclei_RadialDistribution_FracAtD_GOLD_4of4',  # Nuclei_RadialDistribution_FracAtD_GOLD_4of4 on y-axis
```

Comment:
any way to make these labels more dynamic?

Reply:
I think for now I will keep it hardcoded since we likely won't be rerunning these single-models and find different top features.